Training5/src/task-2/src/pages/login-page/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import LoginForm from "../../components/login-form";
+import {useDispatch} from "react-redux";
+import {Navigate, useNavigate} from "react-router-dom";
+import axios from "axios";
+import {base_url} from "../../App";
+import {login} from "../../reducers/auth.slice";
+import {useAppSelector} from "../../hook/useAppSelector";
+
+const LoginPage = () => {
+    const dispatch = useDispatch();
+    const isLogged = useAppSelector((state) => state.root.auth.data.token);
+    const navigate = useNavigate()
+
+    if (isLogged !== '') {
+        return <Navigate replace to='/app'/>
+    }
+
+    const handleLogin = (email: string, password: string) => {
+        const data = {email, password}
+        axios.post(`${base_url}/login`, data).then((res) => {
+            dispatch(login(res.data.token))
+            navigate('/app')
```

Comment:
why do we need it while we have this block code above

```
 if (isLogged !== '') {
        return <Navigate replace to='/app'/>
    }
```

Reply:
`(state) => state.root.auth.data.token` is new function everywhere, why don't you create a selector: getToken, import, and reuse everywhere that you need it